Section themes docs (#45320)
home-assistant/home-assistant.github.io: source/_dashboards/sections.markdown

      Rearranging cards by dragging
    </p>

> ## Applying themes to sections

Reviewer: We can change this title to something more specific:
```suggestion
## Setup of custom themes in specific sections of a view
```

> You can apply different [themes](/integrations/frontend/#defining-themes) to individual sections within a view. This allows you to visually distinguish different areas of your dashboard, such as using warm colors for alerts or cool colors for general information.

Author: The referenced PR is for information only and written as text rather than a link to try to prevent automatic classification (but failed :-) so that the reviewer can check against that. The original documentation PR never made it into "next". We should now merge with current since the functionality is already released.

> You can apply different [themes](/integrations/frontend/#defining-themes) to individual sections within a view. This allows you to visually distinguish different areas of your dashboard, such as using warm colors for alerts or cool colors for general information.

Reviewer: We can add a section for prerequisites to make it clear that custom themes must be added in config file:
```suggestion
You can apply different themes to individual sections within a view. This allows you to visually distinguish different areas of your dashboard, such as using warm colors for alerts or cool colors for general information.

### Prerequisites

Before setting a section theme, you must (create your custom themes in YAML configuration file[(/integrations/frontend/#defining-themes)].
```

### Setting a section theme via the UI

> 1. Open your dashboard in edit mode.

Reviewer: To make it clear:
```suggestion
1. Open your dashboard in edit mode: in the top right of the screen, select the edit {% icon "mdi:edit" %} button.
```

2. On the section you want to theme, select the edit {% icon "mdi:edit" %} button.
3. Select **Edit Section**.
4. Go to the **Settings** tab.
5. Use the **Theme** dropdown to select a theme for this section.
6. Select **Save**.

### Setting a section theme via YAML

Add the `theme` property to a section configuration:

```yaml
views:
  - title: Dashboard
    # View theme
    theme: default-theme  
    type: sections
    sections:
      - type: grid
        # Section overrides view theme
        theme: custom-theme  
        cards:
          - type: weather-forecast
            entity: weather.home
      - type: grid
        # No theme specified - inherits view theme
        cards:
          - type: sensor
            entity: sensor.temperature
```

> ### How section themes work

Reviewer: We could also make this title a bit more specific. For example:
```suggestion
### How settings of section themes work
```

Section themes use CSS cascade to provide flexible theming:

- **Override**: Variables defined in a section theme override the view theme for that section.
- **Inheritance**: Variables not defined in the section theme inherit from the view theme.

> - **Dark mode**: If a section theme defines its own dark mode settings, those are used. Otherwise, the view's dark mode settings apply.

Reviewer: I would remove the bold format, keeping it to UI strings only, and would make tags of items more specific:
```suggestion
- Override of variables: Variables defined in a section theme override the view theme for that section.
- Inheritance of variables: Variables not defined in the section theme inherit from the view theme.
- Override of dark mode settings: If a section theme defines its own dark mode settings, those are used.
- Inheritance of dark mode settings: If a section theme does not define its own dark mode settings, the view's dark mode settings apply.
```

> ### Example: Status dashboard with themed sections

Reviewer: ```suggestion
### YAML example: Add custom themes to a new dashboard
```

This example shows a dashboard with a blue theme for general information and an amber/orange theme for system alerts.

First, define the themes in your `themes.yaml` or `configuration.yaml`:

```yaml
frontend:

>     themes:

Author: updated to use configuration.yaml only

        # Main view theme (blue/info tones)
        main_view:
          primary-color: "#2196f3"
          card-background-color: "#f0f8ff"
          primary-text-color: "#1565c0"
          modes:
            dark:
              card-background-color: "#1a2332"
              primary-text-color: "#90caf9"

        # Alert section theme (amber/orange tones)
        alert_section:
          primary-color: "#ff9800"
          card-background-color: "#fff8f0"
          primary-text-color: "#d84315"
          modes:
            dark:
              card-background-color: "#2d1f1a"

>               primary-text-color: "#ffab91"

Author: already addressed in the latest commit.

      Rearranging cards by dragging
    </p>

> ## Setup of custom themes in specific sections of a view

Reviewer 2: ```suggestion
## Setting a section theme 
```

I don't agree with the change to a long title. Title must short as possible.

Reviewer: Yes indeed, that is one of the "best practices" for titles, but there are more. They should identify as clear as possible the topic of the section. And in this section there is more than the procedure of setting a section theme. There are two procedures (setting a section theme via de UI and via YAML), a description and an example, at least. Of course there are many ways to write the title but we should not use the gerund.

### Prerequisites

Before setting a section theme, you must [create your custom themes in YAML configuration](/integrations/frontend/#defining-themes).

### Setting a section theme via the UI

1. Open your dashboard in edit mode: in the top right of the screen, select the edit {% icon "mdi:edit" %} button.
2. On the section you want to theme, select the edit {% icon "mdi:edit" %} button.
3. Select **Edit Section**.
4. Go to the **Settings** tab.
5. Use the **Theme** dropdown to select a theme for this section.
6. Select **Save**.

### Setting a section theme via YAML

Add the `theme` property to a section configuration:

```yaml
views:
  - title: Dashboard
    # View theme
    theme: default-theme  
    type: sections
    sections:
      - type: grid
        # Section overrides view theme
        theme: custom-theme  
        cards:
          - type: weather-forecast
            entity: weather.home
      - type: grid
        # No theme specified - inherits view theme
        cards:
          - type: sensor
            entity: sensor.temperature
```

> ### How settings of section themes work

Reviewer 2: Not sure we need this part, it's the same logic as view theme.

Section themes use CSS cascade to provide flexible theming:

- Override of variables: Variables defined in a section theme override the view theme for that section.
- Inheritance of variables: Variables not defined in the section theme inherit from the view theme.
- Override of dark mode settings: If a section theme defines its own dark mode settings, those are used.
- Inheritance of dark mode settings: If a section theme does not define its own dark mode settings, the view's dark mode settings apply.

> ### YAML example: Add custom themes to a new dashboard

Reviewer 2: ```suggestion
### YAML example
```
Short title

This example shows a dashboard with a blue theme for general information and an amber/orange theme for system alerts.

First, define the themes in your `configuration.yaml`:

```yaml
frontend:
  themes:
    # Main view theme (blue/info tones)
    main_view:
      primary-color: "#2196f3"
      card-background-color: "#f0f8ff"
      primary-text-color: "#1565c0"
      modes:
        dark:
          card-background-color: "#1a2332"
          primary-text-color: "#90caf9"

    # Alert section theme (amber/orange tones)
    alert_section:
      primary-color: "#ff9800"
      card-background-color: "#fff8f0"
      primary-text-color: "#d84315"
      modes:
        dark:
          card-background-color: "#2d1f1a"
          primary-text-color: "#ffab91"

> ```

Reviewer 2: We don't need to explain this, it's already in the **Prerequisites** section.

            entity: sensor.processor_use
          - type: tile
            entity: sensor.memory_use_percent
      
      # General info section inherits blue theme
      - type: grid
        title: Status & Info
        cards:
          - type: tile
            entity: sun.sun
          - type: tile
            entity: weather.home
```

<p class='img'>
    <img src="/images/dashboards/section-theme-light.png" alt="Dashboard with themed sections in light mode"/>
    Dashboard with section themes in light mode
</p>

<p class='img'>
    <img src="/images/dashboards/section-theme-dark.png" alt="Dashboard with themed sections in dark mode"/>
    Dashboard with section themes in dark mode
</p>

### Best practices for section themes

- **Visual hierarchy**: Use different themes to highlight important sections like security alerts or system status.
- **Consistency**: Limit yourself to 2-3 different themes per view to avoid visual clutter.
- **Dark mode**: Either define dark mode colors in your section themes, or let them inherit from the view theme by not setting color variables.
- **Testing**: Check how your themed sections look on mobile devices where sections stack vertically.

> 

Reviewer 2: Is this needed?

      Rearranging cards by dragging
    </p>

> ## Section themes

Reviewer 3: ```suggestion
## Setting a section theme
```

You can apply different themes to individual sections within a view. This allows you to visually distinguish different areas of your dashboard, such as using warm colors for alerts or cool colors for general information.

### Prerequisites

Before setting a section theme, you must [create your custom themes in YAML configuration](/integrations/frontend/#defining-themes).

> ### Setting a section theme via the UI

Reviewer 3: ```suggestion
### To set a section theme via the UI
```

1. Open your dashboard in edit mode: in the top right of the screen, select the edit {% icon "mdi:edit" %} button.
2. On the section you want to theme, select the edit {% icon "mdi:edit" %} button.
3. Select **Edit Section**.
4. Go to the **Settings** tab.
5. Use the **Theme** dropdown to select a theme for this section.
6. Select **Save**.

> ### Setting a section theme via YAML

Reviewer 3: ```suggestion
### To set a section theme via YAML
```
